src/sql/workbench/parts/notebook/test/electron-browser/cell.test.ts

```diff
@@ -245,6 +245,78 @@ suite('Cell Model', function (): void {
 		should(JSON.stringify(cell.source)).equal(JSON.stringify(['']));
 	});
 
+	test('Should parse metadata\'s hide_input tag correctly', async function (): Promise<void> {
+		let notebookModel = new NotebookModelStub({
+			name: '',
+			version: '',
+			mimetype: ''
+		});
+		let contents: nb.ICellContents = {
+			cell_type: CellTypes.Code,
+			source: ''
+		};
+		let model = factory.createCell(contents, { notebook: notebookModel, isTrusted: false });
+
+		should(model.isCollapsed).be.false();
+		model.isCollapsed = true;
+		should(model.isCollapsed).be.true();
+		model.isCollapsed = false;
+		should(model.isCollapsed).be.false();
+
+		let modelJson = model.toJSON();
+		should(modelJson.metadata.tags).not.be.undefined();
+		should(modelJson.metadata.tags).not.containEql('hide_input');
+
+		contents.metadata = {
+			tags: ['hide_input']
+		};
+		model = factory.createCell(contents, { notebook: notebookModel, isTrusted: false });
+
+		should(model.isCollapsed).be.true();
+		model.isCollapsed = false;
+		should(model.isCollapsed).be.false();
+		model.isCollapsed = true;
+		should(model.isCollapsed).be.true();
+
+		modelJson = model.toJSON();
+		should(modelJson.metadata.tags).not.be.undefined();
+		should(modelJson.metadata.tags).containEql('hide_input');
```

Comment:
it may sound silly, but let's add juust one final condition to tests here... the presence of another tag that isn't 'hide_input'.